### PR TITLE
(MODULES-6751) Convert acceptance tests to beaker rspec

### PR DESCRIPTION
**Pull request by ThoughtCrhyme**
Commit moves the acceptance test files formally in ./tests/acceptance to ./spec/acceptance.
Removes unnecessary setup files, rake shims, and acceptance_tests.sh files.
Converts tests from beaker to beaker-rspec.
Consolidates some of the tests into single files.

New execution of acceptance tests is now 'bundle exec rake beaker'.

### CODE REVIEW

**Comment by glennsarti** [comment]
Cursory look and it looks ok.

**Comment by michaeltlombardi** [comment]
Ran the rake task locally, everything was good. 👍

File: spec/acceptance/basic_dsc_resources/failing_dsc_resources_spec.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          throwmessage    => '#{throw_message}',
+        }
+      }
+    MANIFEST
+
+    error_msg = /Error: PowerShell DSC resource PuppetFakeResource  failed to execute Set-TargetResource functionality with error message: #{throw_message}/
+
+    windows_agents.each do |agent|
+      it 'Applies manifest with one failing resource and one successful resource' do
+        on(agent, puppet('apply'), :stdin => dsc_manifest, :acceptable_exit_codes => 0) do |result|
```
Comment (RandomNoun7) [inline spec/acceptance/basic_dsc_resources/failing_dsc_resources_spec.rb]: Do we want to add a check here as well for the good resource?

File: spec/acceptance/dsc_type/psdesiredstateconfiguration_spec.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      end
+
+      it 'Verify results' do
+        # if this file exists, 'absent' didn't work
+        on(agent, "test -f /cygdrive/c/#{fake_name}", :acceptable_exit_codes => [1])
+      end
+    end
+  end
+
+  before(:all) do
```
Comment (RandomNoun7) [inline spec/acceptance/dsc_type/psdesiredstateconfiguration_spec.rb]: This file doesn't need to set up or tear down the fake resource.

**Pull request merged**